src/main/java/org/opensearch/security/action/onbehalf/CreateOnBehalfOfToken.java

```diff
@@ -0,0 +1,155 @@
+import org.opensearch.security.configuration.ConfigurationRepository;
+import org.opensearch.security.dlic.rest.validation.AbstractConfigurationValidator;
+import org.opensearch.security.privileges.PrivilegesEvaluator;
+import org.opensearch.security.securityconf.impl.CType;
+import org.opensearch.security.securityconf.impl.SecurityDynamicConfiguration;
+import org.opensearch.security.ssl.SecurityKeyStore;
+import org.opensearch.security.ssl.transport.PrincipalExtractor;
+import org.opensearch.security.ssl.util.SSLConfigConstants;
+import org.opensearch.security.support.ConfigConstants;
+import org.opensearch.threadpool.ThreadPool;
+import org.opensearch.client.node.NodeClient;
+import org.opensearch.security.user.User;
+
+import static org.opensearch.security.dlic.rest.support.Utils.addRoutesPrefix;
+
+public class CreateOnBehalfOfToken extends BaseRestHandler {
+
+    private final JwtVendor vendor;
+    private final ThreadPool threadPool;
+
+    public CreateOnBehalfOfToken(final Settings settings, final ThreadPool threadPool) {
+        Settings testSettings =  Settings.builder()
+        .put("signing_key", "1234567890123456")
+        .put("encryption_key", "1234567890123456").build();
+
+        this.vendor = new JwtVendor(testSettings, Optional.empty());
+        this.threadPool = threadPool;
+    }
+
+	@Override
+	public String getName() {
+		return getClass().getSimpleName();
+	}
+
+	@Override
+	public List<Route> routes() {
+		return addRoutesPrefix(
+            ImmutableList.of(
+                    new Route(Method.POST, "/user/onbehalfof")
+            )
+        );
+	}
+
+    @Override
+    protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient client) throws IOException {
+        switch (request.method()) {
+            case POST:
+                return handlePost(request, client);
+            default:
+                throw new IllegalArgumentException(request.method() + " not supported");
+        }
+    }
+
+    private RestChannelConsumer handlePost(RestRequest request, NodeClient client) throws IOException {
+        return new RestChannelConsumer() {
+            @Override
+            public void accept(RestChannel channel) throws Exception {
+                final XContentBuilder builder = channel.newBuilder();
+                BytesRestResponse response;
+                try {
+                    final Map<String, Object> requestBody = request.contentOrSourceParamParser().map();
+                    final String reason = (String)requestBody.getOrDefault("reason", null);
+
+                    final Integer tokenDuration = Optional.ofNullable(requestBody.get("duration"))
+                        .map(value -> (String)value)
+                        .map(Integer::parseInt)
+                        .map(value -> Math.min(value, 72)) // Max duration is 72 hours
+                        .orElse(24); // Fallback to default;
+
+                    final String source = "self-issued";
```

Comment:
If this is cluster name then this should be able to be obtained from the ClusterState which you can get in `createComponents`.

Reply:
In this REST API's case, the token was created by self-issued by user.  If this was generated by a plugin or extension, then it should be marked with the identifier for traceability.  This is good feedback around if we have the correct information in the token for the audit log or other use cases